docs/getting-started/which-permissions-does-codacy-need-from-my-account.md

```diff
@@ -63,15 +63,15 @@ If you log in with GitHub, Codacy requires the following [app permissions](https
       <td>Read & Write</td>
       <td>Codacy sets the status of commits according to the result of code analysis.</td>
     </tr>
-    <tr>
-      <td>Administration</td>
-      <td>Read & Write</td>
-      <td><a href="#why-does-codacy-ask-for-permission-to-create-ssh-keys">Codacy creates an SSH key</a> on the repository to allow cloning and integrating with your repository.</td>
-    </tr>
     <tr>
       <td>Contents</td>
       <td>Read-Only</td>
-      <td>Codacy accesses repository contents to provide faster code coverage analysis and as part of an initiative to use GitHub App tokens instead of SSH keys when cloning repositories for code quality analysis.</td>
+      <td>Codacy uses installation access tokens when integrating with your repositories and cloning them for code quality analysis, and retrieves repository contents to provide faster code coverage analysis.</td>
```

Comment:
Not sure I'm getting the meaning of this sentence. I'm attempting a reword, let me know if it's correct:

```suggestion
      <td>Codacy uses installation access tokens when integrating with your repositories to clone them for code quality analysis and to retrieve their contents for faster code coverage analysis.</td>
```

Reply:
> Codacy uses installation access tokens when integrating with your repositories to clone them for code quality analysis and to retrieve their contents for faster code coverage analysis.

@nicklem This way we are saying that the new Coverage engine also uses installation access tokens... Does it use tokens? 🤔 
I based on the [original sentence](https://github.com/codacy/docs/pull/1931/files#diff-47d9d57a1ef035b843d334b8ceabbfee55e41dc6fe3e45012197b57631db1c28L74), which does not specify if Contents permissions is required to use tokens or something else...

Reply:
Oh, I see. I got the meaning wrong. The fact is that the fragment at the beginning, "Codacy uses installation access tokens", seems to imply that the installation access tokens are used to access contents, but it's the other way around, i.e. the Contents permission is used to access the installation access tokens?

If so, maybe rewording like this can work?

Either:
```
 <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code quality analysis, and to provide faster code coverage analysis.</td>
```

Or:
```
<td>Codacy retrieves repository contents to provide faster code coverage analysis and to retrieve installation access tokens, which are used to integrate with your repositories and clone them for code quality analysis</td>
```

Reply:
I wonder if we could simplify further, avoid mentioning "faster code coverage analysis", and simply group that into a generic "clone them for code analysis":

```
<td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
```

Reply:
I wonder if we could simplify further, avoid mentioning "faster code coverage analysis", and simply group that into a generic "clone them for code analysis":

```suggestion
      <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
```

Reply:
Perhaps we can simplify even further:

```suggestion
      <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
```